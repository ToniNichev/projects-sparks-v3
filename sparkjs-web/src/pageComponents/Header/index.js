import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import styles from './styles.scss';
import GenralPopup from '../../components/GeneralPopup';
import EventsManager from '../../containers/EventsManager';



const Header = ( {title} ) => {

  const { search } = useLocation();
  return (
    <div>
      <div className={styles.wrapper}>      
        <ul>
          <li><Link to={`/home${search}`}>DEVICES</Link></li>
          <li><Link to={`/account${search}`}>ACCOUNT</Link></li>
          <li><Link to={`/setup${search}`}>SETTINGS</Link></li>       
          <li><Link to={`/about${search}`}>ABOUT</Link></li>
          <li><Link to={`/sign-out${search}`}>SIGN OUT</Link></li>
        </ul>
      </div>
      <GenralPopup showPopup={false} />
    </div>
  );
}
export default Header;