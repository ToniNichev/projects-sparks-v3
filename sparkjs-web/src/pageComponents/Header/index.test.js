import React from 'react';
import { shallow } from 'enzyme';
import Header from '.';
import toJson from 'enzyme-to-json';

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    search: '?test=123'
  })
}));


describe('Testing Header component', () => {
    it('renders as expected!!!', () => {

      const myMock = jest.fn();

      const a = new myMock({one: 111});
      
      console.log(myMock.mock.instances);

      const wrapper = shallow(
        <Header title="Title test" />
      );

    expect(myMock.mock.calls.length).toBe(1);
     //expect(toJson(wrapper)).toMatchSnapshot();
   });
});