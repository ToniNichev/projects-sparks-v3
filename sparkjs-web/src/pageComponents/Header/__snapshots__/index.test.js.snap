// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component renders as expected!!! 1`] = `
<div>
  <div>
    <ul>
      <li>
        <Link
          to="/home?test=123"
        >
          DEVICES
        </Link>
      </li>
      <li>
        <Link
          to="/account?test=123"
        >
          ACCOUNT
        </Link>
      </li>
      <li>
        <Link
          to="/setup?test=123"
        >
          SETTINGS
        </Link>
      </li>
      <li>
        <Link
          to="/about?test=123"
        >
          ABOUT
        </Link>
      </li>
      <li>
        <Link
          to="/sign-out?test=123"
        >
          SIGN OUT
        </Link>
      </li>
    </ul>
  </div>
  <Renderer
    showPopup={false}
  />
</div>
`;
